sstables/sstables.cc

```diff
@@ -1513,6 +1513,11 @@ create_sharding_metadata(schema_ptr schema, const dht::decorated_key& first_key,
     auto prange = dht::partition_range::make(dht::ring_position(first_key), dht::ring_position(last_key));
     auto sm = sharding_metadata();
     auto&& ranges = dht::split_range_to_single_shard(*schema, prange, shard).get0();
+    if (ranges.empty()) {
+        auto split_ranges_all_shards = dht::split_range_to_shards(prange, *schema);
+        sstlog.warn("create_sharding_metadata: first_key={}, last_key={}, shard={}, range={}, split_ranges_single_shard={}, split_ranges_all_shards",
```

Comment:
nitpick: a warn() will be visible by end users, so it might be useful to format this message in a way a user could understand this, or at least be able to figure out if it's something he should be worried about or not. You're not even telling the user here *why* you are logging this - something like "range [which range?] has no intersection with this shard", and perhaps what this means and what might have caused this.

Reply:
yes, it is better this way.